lib/src/native/realm_core.dart

```diff
@@ -295,15 +295,18 @@ class _RealmCore {
   RealmObjectHandle? find(Realm realm, int classKey, Object primaryKey) {
     return using((Arena arena) {
       final realm_value = _toRealmValue(primaryKey, arena);
-      final pointer =
-          _realmLib.invokeTryGetPointer(() => _realmLib.realm_object_find_with_primary_key(realm.handle._pointer, classKey, realm_value.ref, nullptr));
+      final pointer = _realmLib.realm_object_find_with_primary_key(realm.handle._pointer, classKey, realm_value.ref, nullptr);
       if (pointer == nullptr) {
         return null;
       }
 
       return RealmObjectHandle._(pointer);
     });
   }
+
+  void removeRealmObject(RealmObject object) {
+    _realmLib.invokeGetBool(() => _realmLib.realm_object_delete(object.handle._pointer));
```

Comment:
I have a silly question about these lambdas - does dart allocate a new one every time or is it using something clever to inline the call in the `invokeGetBool` method? I'm not overly concerned about performance in `removeObject`, but since this is a common pattern employed throughout the dart SDK, I was wondering if it can become a problem.

Reply:
I think these get inlined.